vita3k/https/src/https.cpp

```diff
@@ -236,285 +63,94 @@ std::string get_web_regex_result(const std::string &url, const std::regex &regex
-            return {};
-        }
-
-        accum = static_cast<uint32_t>(accum << 6) | char_value;
-        accum_bits += 6;
-
-        if (accum_bits >= 8) {
-            accum_bits -= 8;
-            md5_bytes[i++] = static_cast<unsigned char>((accum >> accum_bits) & 0xFF);
-        }
-    }
-
-    return convert_md5_bytes_to_str(md5_bytes);
-}
+typedef int (*CurlDownloadCallback)(void *c, long t, long d);
 
 bool download_file(std::string url, const std::string &output_file_path, ProgressCallback progress_callback) {
-    // Get the HEAD of response
-    auto response = get_web_response(url, "HEAD");
-
-    // Check if the response is not empty
-    if (response.empty()) {
-        LOG_ERROR("Failed to get header on url: {}", url);
+    CURL *curl_download = curl_easy_init();
+    if (!curl_download)
         return false;
-    }
 
-    // Check if the response is a redirection
-    if (response.find("HTTP/1.1 302 Found") != std::string::npos) {
-        std::smatch match;
-        // Get the redirection URL from the response header (Location)
-        if (std::regex_search(response, match, std::regex("Location: (https?://[^\\s]+)"))) {
-            url = match[1];
-        } else {
-            LOG_ERROR("No success found redirection location:\n{}", response);
-            return false;
-        }
+    CurlDownloadCallback curl_callback = +[](void *c, long t, long d) {
+        const auto data = (CallbackData *)c;
 
-        // Get the HEAD of response of the redirected url and check if it is not empty
-        response = get_web_response(url, "HEAD");
-        if (response.empty()) {
-            LOG_ERROR("Failed to get header on redirected url: {}", url);
-            return false;
-        }
-    }
+        if (t == 0)
+            return 0; // Ignore if we dont have the total yet
 
-    // Check if the response is resource not found
-    if (response.find("HTTP/1.1 404 Not Found") != std::string::npos) {
-        LOG_ERROR("404 Not Found");
-        return false;
-    }
+        if (!data->second)
+            return 0;
+        const auto progress_percent = static_cast<float>(d) / static_cast<float>(t) * 100.0f;
 
-    // Get the MD5 from the response header (Content-MD5)
-    const auto content_md5 = get_content_md5(response);
-    if (content_md5.empty()) {
-        LOG_ERROR("Failed to get Content-MD5 on header: {}", response);
-        return false;
-    }
+        const auto elapsed_time_ms = std::difftime(get_current_time_ms(), data->first);
 
-    // Get the file size from the response header (Content-Length)
-    const auto file_size = get_file_size(response);
-    if (file_size == 0) {
-        LOG_ERROR("Failed to get file size");
-        return false;
-    }
+        // Calculate remaining time in seconds
+        const auto remaining_bytes = static_cast<double>(t - d);
+        const auto remaining_time = static_cast<uint64_t>((remaining_bytes / d) * elapsed_time_ms) / 1000;
```

Comment:
Why remove my code for get raimaining time ? 
Your code is bad, you need using double obligatory
And no remove my code inside pause state... 
I have no coded all for no reason, all is here for get it good works, here you just broken it